x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -83,25 +86,24 @@ public SecurityActionFilter(AuthenticationService authcService, AuthorizationSer
         if (licenseState.isSecurityEnabled()) {
             final ActionListener<Response> contextPreservingListener =
                     ContextPreservingActionListener.wrapPreservingContext(listener, threadContext);
+            final ActionListener<Response> postActionExecutionListener = ActionListener.delegateFailure(contextPreservingListener,
+                    (ignore, response) -> {
+                        String requestId = AuditUtil.extractRequestId(threadContext);
+                        Authentication authentication = securityContext.getAuthentication();
+                        auditTrailService.get().actionResponse(requestId, authentication, action, request, response);
+                        contextPreservingListener.onResponse(response);
+                    });
```

Comment:
This intercepts the response for the action that is executed inside the rest handler (that do NOT go to other cluster nodes).